### ISSUE TITLE
Allow repeated pipeline steps

### ISSUE THREAD

**Issue by ppinchuk** [NREL/gaps#11]
Copy of NREL/rev#21.

Essentially we want to be able to support multiple repeated steps in a pipeline execution. This would currently break status tracking because the command (step) name is used to key the status dictionary

I.e. we need to be able to support 
```
"pipeline": [
        {
            "generation": "./config_gen.json"
        },
        {
            "collect": "./config_collect.yaml"
        },
        {
            "generation": "./config_other_gen.json"
        },
        {
            "collect": "./config_other_collect.yaml"
        },
...
    ]
```

**Comment by ppinchuk** [comment]
This is quite challenging... The biggest issue is that `Status` dictionaries are currently keyed by the command name, so the infrastructure inherently cannot support duplicate commands. Therefore, we would have to refactors Status to key on something that is definitely unique to each step - i.e. the pipeline step. This means The `Pipeline` class needs to pass the pipeline step to the CLI command callback, which in turn needs to pass it off all the way to the hpc kickoff functions so they can records the jobs as submitted with the pipeline step index. Moreover, the pipe index needs to be passed into the node command itself, since status updates need to know the pipe step. 

This kind of change is guaranteed to break almost all tests. It will also break legacy code. For legacy in particular, we will need to implement a "smart" lookup that only updates the status of commands that have been submitted but not completed yet (the underlying assumption being that legacy pipelines don't allow repeated steps anyways).

One change that has been hinted at above is that we will need to add a `pipeline_step` option (or something like that) to the `Status.mark_job_as_submitted`  function, which will make it more difficult for external users to use this functionality. This may not be too big of a deal (we may not want that to be exposed anyways), but repos like `sup3r` may suffer unnecessarily 

TL;DR This requires a major refactor and is likely not worth it unless someone badly requires this functionality.

**Comment by ppinchuk** [comment]
J/K found a not too-bad workaround
